src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -4814,15 +4869,24 @@ export class ResultsViewPageStore {
      * For annotations of Genome Nexus we want to fetch lazily
      */
     @cached get genomeNexusCache() {
-        return new GenomeNexusCache();
+        return new GenomeNexusCache(
+            fetchFactory(['annotation_summary'], this.genomeNexusClient)
```

Comment:
this fetchFactory is actually very specific to fetchVariantAnnotationsByMutation.   but here in resultsviewpagestore, it would be very hard to understand that.  it sounds like a generic thing.  I would actually change the name fetchFactory to something very specific:    createVariantAnnotationsByMutationFetcher

Reply:
i mean change it where it's defined